intents_consolidator/src/intents/consolidator/handlers/facial_recognition_handler.cpp

```diff
@@ -15,16 +16,32 @@ namespace consolidator
 {
 
 FacialRecognitionResult::FacialRecognitionResult(
-  const machine_intents_interfaces::msg::FacialRecognitionResult * msg)
+  const machine_intents_interfaces::msg::FacialRecognitionResult * msg, const entity::Pose pose)
 {
   embedding_ = std::make_unique<datastructures::Embedding>(msg->embedding);
+  pose_ = pose
 }
 
 
 void FacialRecognitionHandler::Merge(FacialRecognitionResult & res, State & state)
 {
-  //
-  return false;
+  double acceptable_error = 0.02;
+  bool matched = false;
+  for(auto & element : state.Occupants()) {
+    double comparison = res.embedding_.Compare(element.GetEmbedding());
+    if(abs(comparison) < acceptable_error) {
+      matched = true;
+      element.SetEmbedding(std::move(res.GetEmbedding()));
+      element.add_timestamped_pose(time, res.GetPose());
+      break;
```

Comment:
I think you can just return here instead of break. Then, no need for that "if" in the bottom check.

I would add some comments to explain this logic too